### DISCUSSION TITLE
Entity Constructor - Misleading documentation

### DISCUSSION THREAD

**Discussion by ColinCee** [mikro-orm/mikro-orm#2210] [Q&A]
**Describe the bug**

https://mikro-orm.io/docs/entity-constructors/

> Internally, MikroORM never calls entity constructor, so you are free to use it as you wish. The constructor will be called only when you instantiate the class yourself via new operator, so it is a handy place to require your data when creating new entity.

This is not true, and it does call the constructor, see [this line](https://github.com/mikro-orm/mikro-orm/blob/d570a76d3ca14a2c62313929c146c51db6b78a62/packages/core/src/entity/EntityFactory.ts#L95) and my stack trace
**Stack trace**
```
/home/colin/code/service-price-collation/src/entities/Message.ts:49
    this.discordUserId = author.id;
                                ^
TypeError: Cannot read property 'id' of undefined
    at new Message (/home/colin/code/service-price-collation/src/entities/Message.ts:49:33)
    at EntityFactory.createEntity (/home/colin/code/service-price-collation/node_modules/@mikro-orm/core/entity/EntityFactory.js:69:20)
    at EntityFactory.create (/home/colin/code/service-price-collation/node_modules/@mikro-orm/core/entity/EntityFactory.js:34:77)
    at SqlEntityManager.find (/home/colin/code/service-price-collation/node_modules/@mikro-orm/core/EntityManager.js:100:52)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
...
```

**To Reproduce**
1. My entity
```
@Entity()
export class Message {
  @PrimaryKey()
  public id!: number;

  @Property()
  createdAt: Date = new Date();

  @Property({ onUpdate: () => new Date() })
  updatedAt: Date = new Date();

  @Property()
  public discordMessageId: string;

  @Property()
  public discordChannelId: string;

  @Property()
  public discordUserId: string;

  constructor(
    { channelId, id, author }: DiscordMessage,
  ) {
    this.discordChannelId = channelId;
    this.discordMessageId = id;
    this.discordUserId = author.id;
  }
}

```
2. Try to query using repository manager
```
const repo = orm.em.getRepository(Service);
const messages = await repo.findAll();

/// This same applies when doing
const messages = await em.find(Message, {});
```

**Expected behavior**
Returns all the data in the db

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | 16.8 |
| typescript | 4.3.5 |
| mikro-orm | 4.5.9 |
| @mikro-orm/postgresql | 4.5.9 |

**Comment by B4nan** [comment]
`em.find()` won't call entity constructors unless you use `forceEntityConstructor: true` in the config. You must have something wrongly set up if you are not using that flag - it is not about entity definition.
